src/jailer/src/cgroup.rs

```diff
@@ -449,20 +446,25 @@ pub mod test_util {
     use std::io::Write;
     use std::path::{Path, PathBuf};
 
-    use super::PROC_MOUNTS;
+    use utils::rand;
+
+    pub fn get_mock_proc_mounts() -> String {
+        format!(
+            "/tmp/firecracker/test/{}/jailer/proc/mounts",
+            rand::rand_alphanumerics(4).into_string().unwrap()
+        )
+    }
```

Comment:
`get_mock_proc_mounts()` is always called just before `MockCgroupFs::new()`. How about putting `get_mock_proc_monts()` inside the `MockCgroupFs::new()`? By doing this, the lifetime parameter `<'a>` of `MockCgroupFs` can be removed as follows:
```
-    pub fn get_mock_proc_mounts() -> String {
-        format!(
-            "/tmp/firecracker/test/{}/jailer/proc/mounts",
-            rand::rand_alphanumerics(4).into_string().unwrap()
-        )
-    }
-
     #[derive(Debug)]
-    pub struct MockCgroupFs<'a> {
+    pub struct MockCgroupFs {
         mounts_file: File,
-        pub proc_mount_path: &'a str,
+        pub proc_mount_path: String,
     }
 
     // Helper object that simulates the layout of the cgroup file system
     // This can be used for testing regardless of the availability of a particular
     // version of cgroups on the system
-    impl<'a> MockCgroupFs<'a> {
+    impl MockCgroupFs {
         pub fn create_file_with_contents<P: AsRef<Path> + Debug>(
             filename: P,
             contents: &str,
@@ -480,10 +473,12 @@ pub mod test_util {
             Ok(())
         }
 
-        pub fn new(
-            mock_proc_mount: &'a str,
-        ) -> std::result::Result<MockCgroupFs<'a>, std::io::Error> {
-            let mock_proc_dir = Path::new(mock_proc_mount).parent().unwrap();
+        pub fn new() -> std::result::Result<MockCgroupFs, std::io::Error> {
+            let mock_proc_mount = format!(
+                "/tmp/firecracker/test/{}/jailer/proc/mounts",
+                rand::rand_alphanumerics(4).into_string().unwrap()
+            );
+            let mock_proc_dir = Path::new(&mock_proc_mount).parent().unwrap();
```

Reply:
fixed in 540eea53557bdbeef44068e5db60f1c705bd339f and 14f8f38b9819514956eb5b9103ec7cd9cccdd6a1